applications/crossbar/src/modules/cb_port_requests.erl

```diff
@@ -715,6 +721,23 @@ summary(Context) ->
 %% @doc
 %% @end
 %%--------------------------------------------------------------------
+-spec by_number(cb_context:context(), ne_binary()) -> cb_context:context().
+by_number(Context, Number) ->
+    E164 = wnm_util:to_e164(Number),
+    AccountId = cb_context:account_id(Context),
+    ViewOptions = [{'key', [AccountId, E164]}, 'include_docs'],
```

Comment:
I think this should be defined as:

`ViewOptions = [{'keys', summary_keys(Context, E164)}, 'include_docs'],`

where `summary_keys/2` looks like:

```erlang
summary_keys(Context, E164) ->
    summary_keys(Context, E164, props:get_value(<<"accounts">>, cb_context:req_nouns(Context))).

summary_keys(_Context, E164, [AccountId]) ->
    [[AccountId, E164]];
summary_keys(Context, E164, [AccountId, ?DESCENDANTS]) ->
    DescendantIds = descendant_ids(AccountId),
    [[Id, E164] || Id <- [AccountId | DescendantIds]].

descendant_ids(AccountId) ->
    case couch_mgr:get_results(?WH_ACCOUNTS_DB
                               ,<<"accounts/listing_by_descendants">>
                               ,[{'startkey', [AccountId]}
                                 ,{'endkey', [AccountId, wh_json:new()]}
                                ]
                              )
    of
        {'ok', JObjs} -> [wh_json:get_value([<<"value">>, <<"id">>], JObj) || JObj <- JObjs];
        {'error', _E} -> []
    end.
```

This is untested but hopefully conveys the idea?